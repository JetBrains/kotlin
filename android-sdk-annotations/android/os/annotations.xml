<root>
    <item name='android.os.Parcel void readBinderList(java.util.List&lt;android.os.IBinder&gt;) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Parcel void readStringList(java.util.List&lt;java.lang.String&gt;) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.BatteryManager EXTRA_PLUGGED'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Binder boolean transact(int, android.os.Parcel, android.os.Parcel, int) 2'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.WorkSource boolean diff(android.os.WorkSource) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.TokenWatcher void dump(java.io.PrintWriter) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.CountDownTimer android.os.CountDownTimer start()'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Environment MEDIA_BAD_REMOVAL'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Bundle Bundle(android.os.Bundle) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.RecoverySystem void verifyPackage(java.io.File, android.os.RecoverySystem.ProgressListener, java.io.File) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.AsyncTask Result get(long, java.util.concurrent.TimeUnit) 1'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.ParcelFileDescriptor void writeToParcel(android.os.Parcel, int) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Parcel void readTypedArray(T[], android.os.Parcelable.Creator&lt;T&gt;) 1'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.ResultReceiver void writeToParcel(android.os.Parcel, int) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Environment DIRECTORY_MOVIES'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Messenger void writeMessengerOrNullToParcel(android.os.Messenger, android.os.Parcel) 1'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.StrictMode.ThreadPolicy.Builder android.os.StrictMode.ThreadPolicy.Builder permitCustomSlowCalls()'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Parcel void readTypedList(java.util.List&lt;T&gt;, android.os.Parcelable.Creator&lt;T&gt;) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.AsyncTask.Status RUNNING'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.PatternMatcher PatternMatcher(android.os.Parcel) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.BatteryManager EXTRA_LEVEL'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.StrictMode.ThreadPolicy.Builder android.os.StrictMode.ThreadPolicy.Builder penaltyLog()'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.StrictMode.ThreadPolicy.Builder android.os.StrictMode.ThreadPolicy.Builder penaltyDeath()'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.ParcelUuid void writeToParcel(android.os.Parcel, int) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Message android.os.Message obtain(android.os.Message) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Parcel java.util.ArrayList&lt;T&gt; createTypedArrayList(android.os.Parcelable.Creator&lt;T&gt;) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.StrictMode android.os.StrictMode.ThreadPolicy allowThreadDiskReads()'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.UserManager long getSerialNumberForUser(android.os.UserHandle) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Environment MEDIA_CHECKING'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.StrictMode.ThreadPolicy.Builder android.os.StrictMode.ThreadPolicy.Builder detectDiskWrites()'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.ParcelFileDescriptor.AutoCloseOutputStream AutoCloseOutputStream(android.os.ParcelFileDescriptor) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.MemoryFile void writeBytes(byte[], int, int, int) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Parcel void readLongArray(long[]) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.RemoteCallbackList boolean register(E, java.lang.Object) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.RecoverySystem void installPackage(android.content.Context, java.io.File) 1'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Parcel void readTypedList(java.util.List&lt;T&gt;, android.os.Parcelable.Creator&lt;T&gt;) 1'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.StrictMode.VmPolicy.Builder android.os.StrictMode.VmPolicy build()'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.StrictMode.ThreadPolicy.Builder Builder(android.os.StrictMode.ThreadPolicy) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Bundle void putAll(android.os.Bundle) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.AsyncTask android.os.AsyncTask&lt;Params,Progress,Result&gt; execute(Params...)'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.StrictMode.ThreadPolicy.Builder android.os.StrictMode.ThreadPolicy.Builder penaltyFlashScreen()'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Parcel void readStringArray(java.lang.String[]) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.BatteryManager EXTRA_PRESENT'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Parcel void readList(java.util.List, java.lang.ClassLoader) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Handler boolean sendMessageAtFrontOfQueue(android.os.Message) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Message void copyFrom(android.os.Message) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.ParcelFileDescriptor android.os.ParcelFileDescriptor fromFd(int)'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.RemoteCallbackList boolean unregister(E) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Handler boolean sendMessageAtTime(android.os.Message, long) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.MemoryFile java.io.InputStream getInputStream()'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.StrictMode.ThreadPolicy.Builder android.os.StrictMode.ThreadPolicy.Builder detectAll()'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.ParcelFileDescriptor android.os.ParcelFileDescriptor open(java.io.File, int) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.StrictMode.ThreadPolicy.Builder android.os.StrictMode.ThreadPolicy.Builder detectCustomSlowCalls()'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Parcel void readBooleanArray(boolean[]) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Parcel T[] createTypedArray(android.os.Parcelable.Creator&lt;T&gt;) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Debug.MemoryInfo void readFromParcel(android.os.Parcel) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Environment java.io.File getExternalStoragePublicDirectory(java.lang.String)'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.DropBoxManager EXTRA_TIME'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.AsyncTask android.os.AsyncTask&lt;Params,Progress,Result&gt; executeOnExecutor(java.util.concurrent.Executor, Params...)'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Environment MEDIA_UNMOUNTABLE'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Handler boolean sendMessage(android.os.Message) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.BatteryManager EXTRA_TEMPERATURE'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.StrictMode.ThreadPolicy.Builder android.os.StrictMode.ThreadPolicy.Builder permitDiskWrites()'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.DropBoxManager EXTRA_TAG'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Environment MEDIA_UNMOUNTED'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.WorkSource boolean add(android.os.WorkSource) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.AsyncTask android.os.AsyncTask.Status getStatus()'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Parcel android.os.IBinder readStrongBinder()'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.StrictMode.VmPolicy.Builder android.os.StrictMode.VmPolicy.Builder penaltyDeath()'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.StrictMode.ThreadPolicy.Builder android.os.StrictMode.ThreadPolicy.Builder detectDiskReads()'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Messenger void writeToParcel(android.os.Parcel, int) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.MemoryFile int readBytes(byte[], int, int, int) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.WorkSource void writeToParcel(android.os.Parcel, int) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.AsyncTask.Status FINISHED'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Message void writeToParcel(android.os.Parcel, int) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.StrictMode void setVmPolicy(android.os.StrictMode.VmPolicy) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Parcel void readIntArray(int[]) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Handler boolean sendMessageDelayed(android.os.Message, long) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.RecoverySystem void rebootWipeCache(android.content.Context) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Environment DIRECTORY_ALARMS'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Handler Handler(android.os.Looper, android.os.Handler.Callback) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.StrictMode.VmPolicy.Builder android.os.StrictMode.VmPolicy.Builder detectActivityLeaks()'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.ParcelUuid android.os.ParcelUuid fromString(java.lang.String)'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.BatteryManager EXTRA_TECHNOLOGY'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.BatteryManager EXTRA_VOLTAGE'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Environment MEDIA_MOUNTED_READ_ONLY'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.StrictMode android.os.StrictMode.ThreadPolicy allowThreadDiskWrites()'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Process android.os.UserHandle myUserHandle()'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Messenger Messenger(android.os.Handler) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Parcel void readTypedArray(T[], android.os.Parcelable.Creator&lt;T&gt;) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.AsyncTask void execute(java.lang.Runnable) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Environment java.io.File getDataDirectory()'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.StrictMode android.os.StrictMode.ThreadPolicy getThreadPolicy()'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Handler void dump(android.util.Printer, java.lang.String) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.StrictMode android.os.StrictMode.VmPolicy getVmPolicy()'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Environment DIRECTORY_MUSIC'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.StrictMode.VmPolicy.Builder Builder(android.os.StrictMode.VmPolicy) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.ParcelFileDescriptor.AutoCloseInputStream AutoCloseInputStream(android.os.ParcelFileDescriptor) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Environment DIRECTORY_PICTURES'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.UserManager boolean isUserRunningOrStopping(android.os.UserHandle) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.StrictMode.ThreadPolicy.Builder android.os.StrictMode.ThreadPolicy.Builder permitDiskReads()'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Environment MEDIA_SHARED'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.TokenWatcher void acquire(android.os.IBinder, java.lang.String) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.WorkSource boolean remove(android.os.WorkSource) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Looper android.os.MessageQueue myQueue()'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Parcel void readMap(java.util.Map, java.lang.ClassLoader) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.RecoverySystem void rebootWipeUserData(android.content.Context) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Handler void dispatchMessage(android.os.Message) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.StrictMode.VmPolicy.Builder android.os.StrictMode.VmPolicy.Builder setClassInstanceLimit(java.lang.Class, int)'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.BatteryManager EXTRA_HEALTH'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.StrictMode.VmPolicy.Builder android.os.StrictMode.VmPolicy.Builder detectLeakedRegistrationObjects()'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.ParcelFileDescriptor android.os.ParcelFileDescriptor adoptFd(int)'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Environment DIRECTORY_NOTIFICATIONS'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Debug.MemoryInfo java.lang.String getOtherLabel(int)'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.BatteryManager EXTRA_ICON_SMALL'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Environment java.io.File getRootDirectory()'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Parcel void readByteArray(byte[]) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.StrictMode.ThreadPolicy.Builder android.os.StrictMode.ThreadPolicy.Builder penaltyDialog()'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Environment MEDIA_MOUNTED'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.StrictMode.ThreadPolicy.Builder android.os.StrictMode.ThreadPolicy.Builder penaltyDropBox()'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.UserHandle void writeToParcel(android.os.Parcel, int) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.StrictMode.VmPolicy.Builder android.os.StrictMode.VmPolicy.Builder penaltyDropBox()'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.PatternMatcher void writeToParcel(android.os.Parcel, int) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.DropBoxManager ACTION_DROPBOX_ENTRY_ADDED'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Messenger android.os.Messenger readMessengerOrNullFromParcel(android.os.Parcel) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.UserHandle void writeToParcel(android.os.UserHandle, android.os.Parcel) 1'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.DropBoxManager.Entry void writeToParcel(android.os.Parcel, int) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.UserHandle UserHandle(android.os.Parcel) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.StrictMode.VmPolicy.Builder android.os.StrictMode.VmPolicy.Builder detectLeakedSqlLiteObjects()'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.ParcelFileDescriptor android.os.ParcelFileDescriptor fromSocket(java.net.Socket) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Build UNKNOWN'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.BatteryManager EXTRA_SCALE'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Binder boolean onTransact(int, android.os.Parcel, android.os.Parcel, int) 2'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.UserHandle android.os.UserHandle readFromParcel(android.os.Parcel) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.BatteryManager EXTRA_STATUS'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.UserManager boolean isUserRunning(android.os.UserHandle) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Environment MEDIA_REMOVED'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Parcel byte[] createByteArray()'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.StrictMode.ThreadPolicy.Builder android.os.StrictMode.ThreadPolicy.Builder penaltyDeathOnNetwork()'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.StrictMode void setThreadPolicy(android.os.StrictMode.ThreadPolicy) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.StrictMode.VmPolicy.Builder android.os.StrictMode.VmPolicy.Builder detectAll()'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Environment MEDIA_NOFS'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.AsyncTask.Status PENDING'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.RecoverySystem void installPackage(android.content.Context, java.io.File) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.StrictMode.VmPolicy.Builder android.os.StrictMode.VmPolicy.Builder penaltyLog()'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Environment DIRECTORY_PODCASTS'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Handler Handler(android.os.Looper) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Binder android.os.UserHandle getCallingUserHandle()'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.MemoryFile java.io.OutputStream getOutputStream()'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.StrictMode.ThreadPolicy.Builder android.os.StrictMode.ThreadPolicy.Builder detectNetwork()'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Parcel java.lang.String readString()'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Bundle java.util.Set&lt;java.lang.String&gt; keySet()'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Environment DIRECTORY_DCIM'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.ParcelFileDescriptor android.os.ParcelFileDescriptor fromDatagramSocket(java.net.DatagramSocket) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Handler void handleMessage(android.os.Message) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Parcel void readFloatArray(float[]) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.ParcelFileDescriptor android.os.ParcelFileDescriptor[] createPipe()'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Parcel void readDoubleArray(double[]) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.RemoteCallbackList boolean register(E) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Environment DIRECTORY_DOWNLOADS'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.StrictMode.ThreadPolicy.Builder android.os.StrictMode.ThreadPolicy.Builder permitAll()'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Bundle EMPTY'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Binder boolean onTransact(int, android.os.Parcel, android.os.Parcel, int) 1'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.StrictMode.ThreadPolicy.Builder android.os.StrictMode.ThreadPolicy.Builder permitNetwork()'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Bundle void writeToParcel(android.os.Parcel, int) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.AsyncTask android.os.AsyncTask&lt;Params,Progress,Result&gt; executeOnExecutor(java.util.concurrent.Executor, Params...) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Handler java.lang.String getMessageName(android.os.Message) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Parcel byte[] marshall()'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.StrictMode.VmPolicy.Builder android.os.StrictMode.VmPolicy.Builder detectLeakedClosableObjects()'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Bundle java.lang.Object clone()'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Environment DIRECTORY_RINGTONES'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.IBinder boolean transact(int, android.os.Parcel, android.os.Parcel, int) 2'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Parcel void readCharArray(char[]) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Environment java.io.File getDownloadCacheDirectory()'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Parcel void readBinderArray(android.os.IBinder[]) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Bundle void readFromParcel(android.os.Parcel) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Parcel void appendFrom(android.os.Parcel, int, int) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Debug.MemoryInfo void writeToParcel(android.os.Parcel, int) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.StrictMode.ThreadPolicy.Builder android.os.StrictMode.ThreadPolicy build()'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Looper java.lang.Thread getThread()'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.PowerManager android.os.PowerManager.WakeLock newWakeLock(int, java.lang.String)'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
    <item name='android.os.Parcelable void writeToParcel(android.os.Parcel, int) 0'>
        <annotation name='org.jetbrains.annotations.NotNull'/>
    </item>
</root>
