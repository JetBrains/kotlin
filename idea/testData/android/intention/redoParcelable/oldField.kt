// INTENTION_CLASS: org.jetbrains.kotlin.android.intention.RedoParcelableAction
import android.os.Parcel
import android.os.Parcelable

class <caret>SomeData(parcel: Parcel) : Parcelable {
    var count = 0

    init {
        oldField = parcel.readInt()
        field9 = parcel.readByte() != 0.toByte()
        goodArray = parcel.createTypedArray(SuperParcelable)
        goodList = parcel.createTypedArrayList(SuperParcelable)
        parcelableProperty = parcel.readParcelable(SuperParcelable::class.java.classLoader)
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeInt(oldField)
        parcel.writeByte(if (field9) 1 else 0)
        parcel.writeTypedArray(goodArray, flags)
        parcel.writeTypedList(goodList)
        parcel.writeParcelable(parcelableProperty, flags)
    }

    override fun describeContents(): Int {
        return 42
    }

    companion object CREATOR : Parcelable.Creator<SomeData> {
        override fun createFromParcel(parcel: Parcel): SomeData {
            return SomeData(parcel)
        }

        override fun newArray(size: Int): Array<SomeData?> {
            return arrayOfNulls(size)
        }
    }
}