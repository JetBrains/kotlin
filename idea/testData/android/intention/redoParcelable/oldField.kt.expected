// INTENTION_CLASS: org.jetbrains.kotlin.android.intention.RedoParcelableAction
import android.os.Parcel
import android.os.Parcelable

class SomeData() : Parcelable {
    var count = 0

    constructor(parcel: Parcel) : this() {
        count = parcel.readInt()
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeInt(count)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<SomeData> {
        override fun createFromParcel(parcel: Parcel): SomeData {
            return SomeData(parcel)
        }

        override fun newArray(size: Int): Array<SomeData?> {
            return arrayOfNulls(size)
        }
    }

}